### PR TITLE
Tests: Clean up skipped e2e tests

### LINKED ISSUE #15538: Testing: Enable or remove skipped e2e tests
We have recently disabled a few e2e tests to make Travis job pass consistently. Related PRs: #15059, #14986, #15211.

This issue is opened to coordinate efforts to either fix or remove all skipped e2e tests:
- [x] `packages/e2e-tests/specs/block-deletion.test.js`
  - [x] `results in two remaining blocks and positions the caret at the end of the second block` (disabled in #15059), more details in https://github.com/WordPress/gutenberg/pull/15059#issuecomment-484647474:
    > I've spent some time debugging this specific failure. In doing so, I observe the tedium of debugging on behalf of these other popular plugins (and not knowing which specific one could be a culprit). One lead I have is in the fact that you can observe in standard use of the block settings menu, the "Convert to Reusable Block" button [appears only after a short delay](https://github.com/WordPress/gutenberg/blob/58da78ca0c6ba70d3b1860b9fcfbcf1722e9e282/packages/editor/src/components/reusable-blocks-buttons/reusable-block-convert-button.js#L22-L24) (because it fetches user permission to create reusable block). I wonder that there may be a race condition similar to what we've seen previously with animations and the difference in time between when Puppeteer calculates a position and when it enacts a specific interaction (#13739). In this case, I wonder if it calculates the position of "Remove Block", then clicks, but in the time between those two steps the "Add to Reusable Blocks" button [appears in the exact location the "Remove Block" button had been](https://user-images.githubusercontent.com/1779930/56385190-6e2b0500-61ec-11e9-953f-fbcd42fac844.png). This could explain the mismatch in the snapshot where, for no discernible reason, a reusable block exists in the content.
- [ ] `packages/e2e-tests/specs/embedding.test.js`
  - [ ] `should transform from video to embed block when YouTube URL is pasted` (disabled in #15059)
  - [ ] `should transform from image to embed block when Instagram URL is pasted` (disabled in #15059)
  - [ ] `should transform from audio to embed block when Soundcloud URL is pasted` (disabled in #15059)
- [ ]  `packages/e2e-tests/specs/links.test.js`
  - [ ] `allows autocomplete suggestions to be selected with the mouse` (old issue #13044, previous attempt to fix #13161, disabled in #15059)
  - [ ] `allows autocomplete suggestions to be navigated with the keyboard` (disabled in #15211)
- [ ]  `packages/e2e-tests/specs/nux.test.js` - the following tests were never enabled even though blocking issues were resolved, they should be removed if they no longer are relevant
  - [ ] `should show tips as disabled if all tips have been shown`
  - [ ] `should reset tips if all tips have been shown and show tips was unchecked`
  - [ ] `should show tips if "Show tips" was disabled on a draft and then enabled`
- [ ] `packages/e2e-tests/specs/plugins/wp-editor-meta-box.test.js ` (disabled in #15211), more info in https://github.com/WordPress/gutenberg/pull/15211#issuecomment-487091562:
  > sounds familiar to issues I'd seen in the past where sometimes TinyMCE takes some time to initialize, so "click" in the test to activate the field might be delayed. It could possibly be improved by ensuring that whatever DOM changes might occur to indicate that the initialization is complete are accounted for in e.g. `page.waitForSelector`.
- [ ] `packages/e2e-tests/specs/taxonomies.test.js`
  - [ ] `should be able to open the tags panel and create a new tag if the user has the right capabilities`  (disabled in #15211), more details in https://github.com/WordPress/gutenberg/pull/15211#issuecomment-487091562: 
    > sounds quite similar to issues we'd had recently with the need for this permissions information to be fetched from the REST API: #15061
    
    Given that taxonomies are loaded in the sidebar and aren't mandatory on the first load of the editor, it might be enough to refactor test to ensure that it waits until data is loaded with REST API.

## Reproducing issues

The biggest challenge we encountered in the past was the fact that all the aforementioned tests were failing randomly on Travis but it was nearly impossible to reproduce it locally. There is a plan to add some debugging capabilities to `npm run test-e2e` command but for the time being, I recommend using the following patch to slow down CPU and network to simulate conditions similar to Travis. You still might need to run the same test a few times to be able to see failure.

```diff
diff --git a/packages/e2e-tests/config/setup-test-framework.js b/packages/e2e-tests/config/setup-test-framework.js
index b14717620..f8e70c62e 100644
--- a/packages/e2e-tests/config/setup-test-framework.js
+++ b/packages/e2e-tests/config/setup-test-framework.js
@@ -198,6 +198,20 @@ async function runAxeTestsForBlockEditor() {
 // other posts/comments/etc. aren't dirtying tests and tests don't depend on
 // each other's side-effects.
 beforeAll( async () => {
+	const client = await page.target().createCDPSession();
+	await client.send( 'Network.emulateNetworkConditions', {
+		// Whether chrome should simulate
+		// the absence of connectivity
+		offline: false,
+		// Simulated download speed (bytes/s)
+		downloadThroughput: 1.5 * 1024 * 1024 / 8,
+		// Simulated upload speed (bytes/s)
+		uploadThroughput: 750 * 1024 / 8,
+		// Simulated latency (ms)
+		latency: 40,
+	} );
+	await client.send( 'Emulation.setCPUThrottlingRate', { rate: 8 } );
+
 	capturePageEventsForTearDown();
 	enablePageDialogAccept();
 	observeConsoleLogging();
```

It is super useful to run e2e tests in watch mode and with the real browser:

```sh
npm run test-e2e:watch -- --puppeteer-interactive --puppeteer-slowmo=0
```

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Closes #15538.

## Removed `npm run test-e2e:local`

As part of this PR, I also removed `pretest-e2e` and `test-e2e:local` scripts since we agreed that they aren't necessary.

## Removed tests

It removes the following skipped e2e tests which no longer works and it might be too much effort to bring them up to date. They were disabled at least since May so I don't think we really need them:

- [x] `packages/e2e-tests/specs/embedding.test.js`
  - [x] `should transform from video to embed block when YouTube URL is pasted` (disabled in #15059)
  - [x] `should transform from image to embed block when Instagram URL is pasted` (disabled in #15059)
  - [x] `should transform from audio to embed block when Soundcloud URL is pasted` (disabled in #15059)
- [x]  `packages/e2e-tests/specs/links.test.js`
  - [x] `allows autocomplete suggestions to be selected with the mouse` (old issue #13044, previous attempt to fix #13161, disabled in #15059)
  - [x] `allows autocomplete suggestions to be navigated with the keyboard` (disabled in #15211)
- [x]  `packages/e2e-tests/specs/nux.test.js` - the following tests were never enabled even though blocking issues were resolved, they should be removed if they no longer are relevant
  - [x] `should show tips as disabled if all tips have been shown`
  - [x] `should reset tips if all tips have been shown and show tips was unchecked`
  - [x] `should show tips if "Show tips" was disabled on a draft and then enabled`

## Fixed tests

I was able to fix 2 tests which cover more important functionality following the comments shared in the parent issue:

- [x] `packages/e2e-tests/specs/plugins/wp-editor-meta-box.test.js ` (disabled in #15211), more info in https://github.com/WordPress/gutenberg/pull/15211#issuecomment-487091562:
  > sounds familiar to issues I'd seen in the past where sometimes TinyMCE takes some time to initialize, so "click" in the test to activate the field might be delayed. It could possibly be improved by ensuring that whatever DOM changes might occur to indicate that the initialization is complete are accounted for in e.g. `page.waitForSelector`.

- [x] `packages/e2e-tests/specs/taxonomies.test.js`
  - [x] `should be able to open the tags panel and create a new tag if the user has the right capabilities`  (disabled in #15211), more details in https://github.com/WordPress/gutenberg/pull/15211#issuecomment-487091562: 
    > sounds quite similar to issues we'd had recently with the need for this permissions information to be fetched from the REST API: #15061

## Testing

Ensure there are no more skipped e2e tests.

`npm run e2e-tests` should always pass.

It might also help to run tests with the real browser as follows and then pick only the tests from the list:
`npm run test-e2e:watch -- --puppeteer-interactive --puppeteer-slowmo=0`

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/taxonomies.test.js [hunk 1]
```diff
@@ -141,8 +139,10 @@ describe( 'Taxonomies', () => {
 		// Click the tag input field.
 		await tagInput.click();
 
+		const tagName = 'tag-' + random( 1, Number.MAX_SAFE_INTEGER );
```
Comment (gziolo) [inline packages/e2e-tests/specs/editor/various/taxonomies.test.js]: @youknowriad - this is the test which pollutes the database. We probably should improve the logic which cleans up the database by removing tags and categories in addition to posts and pages.

**Comment by gziolo** [comment]
I see Travis doesn’t like my changes applied to one of the helpers 😅

**Comment by gziolo** [comment]
> I see Travis doesn’t like my changes applied to one of the helpers 😅

It's all sorted out now 😍

File: packages/e2e-tests/specs/editor/various/taxonomies.test.js (hunk 1 above)
Comment (youknowriad) [reply packages/e2e-tests/specs/editor/various/taxonomies.test.js]: yeah, we can also add an `afterAll` step in this test to clean things.

**Review by youknowriad — APPROVED** [review]
Nice cleanup

**Comment by gziolo** [comment]
I restarted e2e tests 5 times and they passed 100% times for the updated tests 🎉 

I saw only a single test failure for block transforms - this is a knows issue.

**Pull request merged**